src/morphology.cpp

```diff
@@ -63,3 +66,39 @@ void section_geometry::segment(double dx) {
     points = std::move(sampled);
     length = sampled_length;
 }
+
+void section_geometry::compute_length() {
```

Comment:
I'm writing 'right now' a method that sits on the `morphology` class that constructs `section_geometry` structs with the correct fields from e.g. a list of `section_point`s.

I could then remove the `compute_length()` method, and leave it to `morphology` to compute this. Users then are not expected to fill in all the bits of a `section_geometry`, but if they start fiddling with the innards, then on their heads be it. This would be my preferred solution.